### PR TITLE
Do not ICE in `if` without `else` in `async fn`

### LINKED ISSUE #66387: Compiler crash on "if" without "else" in async fn
```rust
async fn f() -> Result<(), failure::Error> {
    if "a" != "b" {
        return Err("c".to_string());
    }
}

```

([Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=7c50ae4cb0d4169c289cbdeea491a48b))

Triggers on playground on beta and nightly channels, not on stable.

Stacktrace: https://gist.github.com/guzt/32437a394c5c3401f8ec80a6863efe56

### PR DESCRIPTION
**Pull request by estebank**
Fix #66387.

### CODE REVIEW

File: src/librustc_typeck/check/generator_interior.rs [hunk 1]
```diff
@@ -244,7 +244,13 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
         // can be reborrowed without needing to spill to a temporary.
         // If this were not the case, then we could conceivably have
         // to create intermediate temporaries.)
-        let ty = self.fcx.tables.borrow().expr_ty(expr);
-        self.record(ty, scope, Some(expr), expr.span);
+        //
+        // The type table might not have invormation for this expression
```
Comment (gilescope) [inline src/librustc_typeck/check/generator_interior.rs]: invormation - typo?

**Review by nikomatsakis — APPROVED** [review]

**Comment by jplatte** [comment]
Isn't the error message backwards in the ui test? Shouldn't it be "expected i32", "found ()"?

**Comment by pnkfelix** [comment]
[beta-accepted](https://zulip-archive.rust-lang.org/131828tcompiler/21461weeklymeeting2019111454818.html#180739560) (last week, sorry for delay!)

**Comment by pnkfelix** [comment]
stable-nominating on basis on my (unconfirmed) hypothesis that this PR fixes #66618.

**Pull request merged**